astropy/cosmology/realizations.py

```diff
@@ -8,53 +8,47 @@
 from astropy.utils.state import ScienceState
 
 from . import parameters
-from .core import Cosmology, FlatLambdaCDM, LambdaCDM
+from .core import Cosmology
 
 __all__ = ["default_cosmology"] + parameters.available
 
 __doctest_requires__ = {"*": ["scipy"]}
 
 # Pre-defined cosmologies. This loops over the parameter sets in the
-# parameters module and creates a LambdaCDM or FlatLambdaCDM instance
-# with the same name as the parameter set in the current module's namespace.
-
+# parameters module and creates a cosmology instance with the same name as the
+# parameter set in the current module's namespace.
+_subclasses = dict(Cosmology.__all_subclasses__())
```

Comment:
I'd prefer a short function here instead of defining a dunder method on `Cosmology`. 

For follow-up, it may make sense to keep a dict `COSMOLOGIES` which gets filled in `Cosmology.__init_subclass__`; that would follow precedent from representations and time formats.

Reply:
> I'd prefer a short function here instead of defining a dunder method on Cosmology.

Sounds good. I can move it off.

> For follow-up, it may make sense to keep a dict COSMOLOGIES which gets filled in Cosmology.__init_subclass__; that would follow precedent from representations and time formats.

This one is trickier because it would actually need to be a tree structure, not a flat dictionary, because this will also be used for I/O and ``LambdaCDM.read`` should be able to read in for a ``FlatLambdaCDM`` file, but not for a ``FLRW`` cosmology.

Reply:
All the more reason to leave it for follow-up!